### PR TITLE
Add community contribution UI elements

### PR DESCRIPTION
**Pull request by bradystroud**

### CODE REVIEW

File: src/components/header/header.js [hunk 1]
```diff
@@ -35,30 +36,39 @@ const Header = ({ displayActions, ruleUri }) => {
...
-            <div></div>
-          )}
+          <div className="action-btn-container">
+            <Auth0Provider
+              domain={process.env.DOMAIN}
+              clientId={process.env.CLIENT_ID}
+              redirectUri={process.env.CALLBACK}
+            >
+              <SignIn />
+            </Auth0Provider>
```
Comment (AndreasLengkeek) [inline src/components/header/header.js:46]: Move this provider to wrap the whole app in the Layout.js file

File: src/components/header/header.js [hunk 2]
```diff
@@ -35,30 +36,39 @@ const Header = ({ displayActions, ruleUri }) => {
...
-              </a>
-            </div>
-          ) : (
-            <div></div>
-          )}
+          <div className="action-btn-container">
+            <Auth0Provider
+              domain={process.env.DOMAIN}
+              clientId={process.env.CLIENT_ID}
+              redirectUri={process.env.CALLBACK}
```
Comment (AndreasLengkeek) [inline src/components/header/header.js:43]: Make sure these variables are in the template file

Comment (AndreasLengkeek) [inline src/components/header/header.js:43]: I would use window.origin.location to get the current environments domain

File: src/templates/rule.js [hunk 3]
```diff
@@ -214,22 +216,7 @@ const Rule = ({ data, location }) => {
...
-              <p className="preview">
-                <span>
-                  8 <FontAwesomeIcon icon={faThumbsUp} className="good" />
-                </span>{' '}
-                <span>
-                  <FontAwesomeIcon icon={faThumbsDown} className="bad" /> 2
-                </span>
-              </p>
-              <p>
-                <small className="suggestion">
```
Comment (AndreasLengkeek) [inline src/templates/rule.js:227]: Leave this outside the React element

File: src/components/reaction/reaction.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { faThumbsUp, faThumbsDown } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+const Reaction = () => {
+  const [likesCount, setLikesCount] = useState(0);
+  const [dislikesCount, setDisikesCount] = useState(0);
+  const [currentReactionType, setCurrentReactionType] = useState(null);
+  var isLoggedIn = true;
```
Comment (AndreasLengkeek) [inline src/components/reaction/reaction.js:9]: if you import `auth0-react` class you can use the loggedIn flag
```const { loggedIn } = useAuth0();```

File: src/components/signin/signin.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      {user && <Profile />}
+      {!isAuthenticated ? (
+        <button className="btn btn-red" onClick={loginWithRedirect}>
+          Log in
+        </button>
+      ) : (
+        <button
+          className="btn btn-red"
+          onClick={() => {
+            logout({ returnTo: 'http://localhost:3000' });
```
Comment (AndreasLengkeek) [inline src/components/signin/signin.js:27]: needs to be generic as this needs to work on multiple environments

File: src/components/signin/signin.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import { useAuth0, Auth0Provider } from '@auth0/auth0-react';
+import React from 'react';
+
+const SignIn = () => {
+  const { isAuthenticated, loginWithRedirect, logout, user } = useAuth0();
+  user && console.log(user);
```
Comment (AndreasLengkeek) [inline src/components/signin/signin.js:6]: remove console logs in production code

**Review by AndreasLengkeek — CHANGES_REQUESTED** [review]

File: src/templates/rule.js [hunk 7]
```diff
@@ -215,12 +217,22 @@ const Rule = ({ data, location }) => {
...
-              <p className="preview">
+              <p>
                 <span>
-                  8 <FontAwesomeIcon icon={faThumbsUp} className="good" />
-                </span>{' '}
+                  <div className="likes-counter-container">{likesCount}</div>
+                  <FontAwesomeIcon
+                    icon={faThumbsUp}
+                    className="good"
+                    onClick={() => setLikesCount(likesCount + 1)}
```
[the remark this answers is not in the document]
Comment (bradystroud) [reply src/templates/rule.js]: I am not actually incrementing `likesCount`. I am updating the state of `likesCount` to one more than the current state. I followed these docs to add this [https://reactjs.org/docs/hooks-state.html](https://reactjs.org/docs/hooks-state.html)

**Review by AndreasLengkeek — APPROVED** [review]

**Pull request merged**